src/FileDropper/FileDropper.tsx

```diff
@@ -0,0 +1,193 @@
+import { CloudUploadIcon } from 'lucide-react';
+
+import {
+  Alert,
+  Box,
+  LinearProgress,
+  Stack,
+  SxProps,
+  Typography,
+  useTheme,
+} from '@mui/material';
+
+import { DndProvider, DropTargetMonitor, useDrop } from 'react-dnd';
+import { HTML5Backend, NativeTypes } from 'react-dnd-html5-backend';
+
+import UploadFileButton, {
+  UploadFileButtonProps,
+} from '../buttons/UploadFileButton/UploadFileButton';
+
+export type FileDropperProps = {
+  id?: string;
+  sx?: SxProps;
+  /**
+   * Callback on files dropped
+   */
+  onDrop?: (files: File[]) => void;
+  /**
+   * Callback on files selected
+   */
+  onChange: UploadFileButtonProps['onChange'];
+  /**
+   * Error to show
+   */
+  error?: string;
+  /**
+   * Smaller text to show, such as limits
+   */
+  hints?: string;
+  /**
+   * Text of the browsing button
+   */
+  buttonText?: string;
+  /**
+   * additional buttons to display next to browse files
+   */
+  buttons?: JSX.Element;
+  /**
+   * When true, allows many files to be selected when browsing
+   */
+  multiple?: UploadFileButtonProps['multiple'];
+  /**
+   * Text displayed by default
+   */
+  message?: string;
+  /**
+   * Text displayed on drag enter
+   */
+  releaseText?: string;
+  /**
+   * Whether files are getting uploaded
+   */
+  isLoading?: boolean;
+  /**
+   * If `isLoading` is true, shows given `upload progress` value in a progress bar
+   */
+  uploadProgress?: number;
```

Comment:
Would it be possible to combine these two props into a single one ? 
If using `uploadProgress` only you could derive a local variable `isLoading` that would be true when uploadProgress is not undefined.